kernel-rpm-script

```diff
@@ -0,0 +1,302 @@
+	--flavor)
+	    flavor="$2"
+	    shift 2
+	    ;;
+	--variant)
+	    flavor="$2"
+	    shift 2
+	    ;;
+
+	--usrmerged)
+	    usrmerged="$2"
+	    shift 2
+	    ;;
+	--image)
+	    image="$2"
+	    shift 2
+	    ;;
+	--systemmap)
+	    systemmap="$2"
+	    shift 2
+	    ;;
+
+	--modulesdir)
+	    modulesdir="$2"
+	    shift 2
+	    ;;
+	--basepackage)
+	    basepackage="$2"
+	    shift 2
+	    ;;
+	--certs)
+	    certs="$2"
+	    shift 2
+	    ;;
+	*) break
+	    ;;
+    esac
+done
+
+wm2=/usr/lib/module-init-tools/weak-modules2
+nvr="$name"-"$version"-"$release"
+
+trigger_purge_kernels() {
+    [ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo Triggering purge-kernels > /dev/kmsg
+    touch /boot/do_purge_kernels
+}
+
+run_wm2() {
+    [ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo wm2 "$@" > /dev/kmsg
+    /bin/bash -${-/e/} $wm2 "$@"
+}
+
+message_install_bl () {
+    echo "You may need to setup and install the boot loader using the"
+    echo "available bootloader for your platform (e.g. grub, lilo, zipl, ...)."
+}
+
+run_bootloader () {
+    if [ -f /etc/sysconfig/bootloader ] &&
+	[ -f /boot/grub/menu.lst -o \
+	-f /etc/lilo.conf      -o \
+	-f /etc/elilo.conf     -o \
+	-f /etc/zipl.conf      -o \
+	-f /etc/default/grub    ]
+	then
+	    return 0
+	else
+	    return 1
+    fi
+}
+
+[ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || \
+    echo "$op" name: "$name" version: "$version" release: "$release" \
+    kernelrelease: "$kernelrelease" flavor: "$flavor" variant: "$variant" \
+    usrmerged: "$usrmerged" image: "$image" systemmap: "$systemmap" \
+    modulesdir: "$modulesdir" basepackage: "$basepackage" certs: "$certs" > /dev/kmsg
+
+script_rc=0
+
+case $op in
+    pre)
+	# see bug #259303
+	# this script runs when the kernel gets updated with YaST
+	# YaST calls rpm always with -U
+	# -U replaces all packages with the new one
+	# rpm removes the files from the old packages after the postinstall script ran
+	# this will double the required space below /boot
+	# remove the files from the old packages to make room for the new initrd
+	# rpm may complain about low disk space if /boot/vmlinux does not fit
+	if [ "$basepackage" = 1 -a "$YAST_IS_RUNNING" != "" ]; then
+	    mydf="$( POSIXLY_CORRECT=1 df -P /boot/ | awk '/^(\/|-[[:blank:]])/{ print $4}' )"
+	    if test "$mydf" != "" ; then
+		echo "Free diskspace below /boot: $mydf blocks"
+		# echo "512 byte blocks: $(( 2 * 1024 * 20 ))"
+		if test "$mydf" -lt  "40960" ; then
+		    echo "make room for new kernel '"$flavor"' because there are less than 20MB available."
+		    # disabled because it breaks patch rpms
+		    #rm -fv /boot/"$image"-*-"$flavor"
+		    rm -fv /boot/initrd-*-"$flavor"
+		fi
+	    fi
+	fi
+
+
+	# On AArch64 we switched from 64k PAGE_SIZE to 4k PAGE_SIZE. Unfortunately
+	# btrfs can only use file systems created with the same PAGE_SIZE. So we
+	# check if the user has any btrfs file systems mounted and refuse to install
+	# in that case.
+	if [ $( uname -m ) = aarch64 -a \
```

Comment:
I think we should put "hacks" like this into a "plugin" directory in a later step.

Reply:
I am not opposed to restructuring this later so long as the interface - API that is provided now is future-proof for such restructuring. In this patch I want to import something that is really close to what was removed from the kernel.

Reply:
Right. No objections. I just typed the thoughts that went through my head when I read this.